universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -17,6 +26,45 @@
  */
 public abstract class BaseHtmlView {
 
+  public Tag renderHeader(String headerText) {
+    return h1(headerText);
+  }
+
+  protected ImmutableList<DomContent> inputWithLabel(
+      String labelValue, String inputId, Optional<String> value) {
+    Tag labelTag = label(labelValue).attr("for", inputId);
+    Tag inputTag = input().withType("text").withId(inputId).withName(inputId);
+    if (value.isPresent()) {
+      inputTag.withValue(value.get());
+    }
+    return ImmutableList.of(labelTag, br(), inputTag, br(), br());
+  }
+
+  public ImmutableList<DomContent> textAreaWithLabel(
+      String labelValue, String inputId, Optional<String> value) {
+    Tag labelTag = label(labelValue).attr("for", inputId);
+    Tag textAreaTag = textarea(value.orElse("")).withType("text").withId(inputId).withName(inputId);
+    return ImmutableList.of(labelTag, br(), textAreaTag, br(), br());
+  }
+
+  public ImmutableList<DomContent> formSelect(
+      String labelValue,
+      String selectId,
+      String[] optionLabels,
+      String[] optionValues,
```

Comment:
Suggestion (blocking): use `ImmutableList` rather than a raw Java array. We should generally avoid using raw arrays unless a library forces us to.

Reply:
Done